website/src/Pages/GalleryPage/SubmissionDetailsPage.jsx

```diff
@@ -0,0 +1,317 @@
+import {
+  Box, Typography, Select, MenuItem, TextField, Button, FormControl,
+} from '@mui/material';
+import axios from 'axios';
+import { React, useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+
+function SubmissionDetailsPage() {
+  const location = useLocation();
+  const {
+    id,
+  } = location.state;
+
+  const [details, setDetails] = useState({
+    mediaData: [''],
+    status: '',
+    title: '',
+    uploader: '',
+    description: '',
+    mediaTypes: [''],
+    tags: [''],
+    date: '',
+    comments: '',
+  });
+  const [edit, setEdit] = useState(false);
+  const [change, setChange] = useState(false);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [action, setAction] = useState('');
+  const [comments, setComments] = useState('');
+  const [update, setUpdate] = useState(0);
+
+  const getSubmissionDetails = async () => {
+    const artwork = await axios.get('http://localhost:4000/submissions/getartwork', { params: { id } });
+    setDetails({
+      mediaData: artwork.data.MediaData,
+      status: artwork.data.Submission.status,
+      title: artwork.data.Submission.title,
+      uploader: artwork.data.Submission.name,
+      email: artwork.data.Submission.email,
+      description: artwork.data.Submission.description,
+      mediaTypes: artwork.data.Submission.mediaTypes,
+      tags: artwork.data.Submission.tags,
+      date: artwork.data.Submission.date,
+      comments: artwork.data.Submission.comments,
+    });
+    setTitle(artwork.data.Submission.title);
+    setDescription(artwork.data.Submission.description);
+    if (artwork.data.Submission.status === 'Incomplete') {
+      setChange(true);
+      setAction(artwork.data.Submission.status);
+    } else if (artwork.data.Submission.status === 'Approved') {
+      setAction('Approve');
+    } else {
+      setAction('Reject');
+    }
+    setComments(artwork.data.Submission.comments);
+  };
+
+  useEffect(() => {
+    getSubmissionDetails();
+  }, [update]);
```

Comment:
is re-retrieving the entire Submission anytime admin updates a good idea, or can we not just work with whatever they inputted when they updated the submission?